plugins/modules/win_gpo_link.ps1

```diff
@@ -0,0 +1,278 @@
+    }
+    # test if glink is not null or only containes white space before continuing.
+    if (!((($gplink.gplink) -like "") -or (($gplink.gplink) -like " "))) {
+        #set variale $o to define link order
+        $o = 0
+        #we split the gplink string in order to seperate the diffent GPO linked
+        $split = $gplink.gplink.split("]")
+        #we need to do a reverse for to get the proper link order
+        for ($s = $split.count - 1; $s -gt -1; $s--) {
+            #since the last character in the gplink string is a "]" the last split is empty we need to ignore it
+            if ($split[$s].length -gt 0) {
+                $o++
+                $order = $o
+                $gpoguid = $split[$s].substring(12, 36)
+                $gpodomainDN = ($split[$s].substring(72)).split(";")
+                $domain = ($gpodomaindn[0].substring(3)).replace(",DC=", ".")
+                $checkdc = (get-addomaincontroller -domainname $domain -discover).name
+                #we test if the $gpoguid is a valid GUID in the domain if not we return a "Oprhaned GpLink or External GPO" in the $gponname
+                $mygpo = get-gpo -guid $gpoguid -domain $domain -server "$($checkdc)$($domain)$null"
+                if ($null -ne $mygpo ) {
+                    $gponame = $MyGPO.displayname
+                    $gpodomain = $domain
+                } else {
+                    $gponame = "Orphaned GPLink"
+                    $gpodomain = $domain
+                }
+                #we test the last 2 charaters of the split do determine the status of the GPO link
+                if (($split[$s].endswith(";0"))) {
+                    $enforced = "No"
+                    $enabled = "Yes"
+                } elseif (($split[$s].endswith(";1"))) {
+                    $enabled = "No"
+                    $enforced = "No"
+                } elseif (($split[$s].endswith(";2"))) {
+                    $enabled = "Yes"
+                    $enforced = "Yes"
+                } elseif (($split[$s].endswith(";3"))) {
+                    $enabled = "No"
+                    $enforced = "Yes"
+                }
+                #we create an object representing each GPOs, its links status and link order
+                $return = New-Object psobject
+                $return | Add-Member -membertype NoteProperty -Name "Target" -Value $target
+                $return | Add-Member -membertype NoteProperty -Name "GPOID" -Value $gpoguid
+                $return | Add-Member -membertype NoteProperty -Name "DisplayName" -Value $gponame
+                $return | Add-Member -membertype NoteProperty -Name "Domain" -Value $gpodomain
+                $return | Add-Member -membertype NoteProperty -Name "Enforced" -Value $enforced
+                $return | Add-Member -membertype NoteProperty -Name "Enabled" -Value $enabled
+                $return | Add-Member -membertype NoteProperty -Name "Order" -Value $order
+                $return
```

Comment:
This is quite an old style of creating a PSObject in PowerShell, it's a lot more idiomatic to do

```suggestion
                [PSCustomObject]@{
                    Target = $target
                    GPOID = $gpoguid
                    DisplayName = $gponame
                    Domain = $gpodomain
                    Enforced = $enforced
                    Enabled = $enabled
                    Order = $order
                }
```